pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -223,12 +223,21 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 func (p Provider) DestroyBootstrap(dir string) error {
 	logrus.Warnf("Destroying GCP Bootstrap Resources")
 	metadata, err := metadata.Load(dir)
+	ctx := context.Background()
 	if err != nil {
 		return err
 	}
-	if err := gcp.DestroyStorage(context.Background(), metadata.ClusterID); err != nil {
+	if err := gcp.DestroyStorage(ctx, metadata.ClusterID); err != nil {
 		return fmt.Errorf("failed to destroy storage")
 	}
+
+	if metadata.GCP == nil {
```

Comment:
nit: Maybe this should be moved up to just below the metadata loading.

Reply:
Fixed.